geo-types/src/point.rs

```diff
@@ -409,6 +414,79 @@ where
     }
 }
 
+#[cfg(test)]
+impl<T> RelativeEq for Point<T>
```

Comment:
Please put the implementation on `Coordinate` and change `Point`'s implementation to delegate to that - like:
```
impl<T> RelativeEq for Point<T> {
    fn relative_eq(self, other, epsilon) -> bool {
        self.0.relative_eq(other.0, epsilon)
    }
   ...
}
```

Reply:
> please put the implementation on 

I think when I submit a fix... there is going to be a radical change here. So I want to signal it ahead of time

in the main issue .. I talked about euclidean distance versus 'Manhattan' distance. 

Coordinate{x: T, y: T} has such a similar structure to Complex{ re: T, im: T}

I think I made a over-complication when I look in approx-0.4..0 at the implementation of Complex I see this ..

The implementation forgets the euclidean distance and just ensures that the both re and im meet the constraint individually. Because I copy/modifed the default implementation .. the same infinity checks for example will be done in the same order .. etc  

```
#[cfg(feature = "num-complex")]
impl<T: RelativeEq> RelativeEq for Complex<T>
where
    T::Epsilon: Clone,
{
    #[inline]
    fn default_max_relative() -> T::Epsilon {
        T::default_max_relative()
    }

    #[inline]
    fn relative_eq(
        &self,
        other: &Complex<T>,
        epsilon: T::Epsilon,
        max_relative: T::Epsilon,
    ) -> bool {
        T::relative_eq(&self.re, &other.re, epsilon.clone(), max_relative.clone())
            && T::relative_eq(&self.im, &other.im, epsilon.clone(), max_relative.clone())
    }
}
```

Reply:
Honestly - I think using manhattan distance is fine — it might even be preferred because it's so simple.

Reply:
Great -- so this has been fixed. The implementation was not moved into coords .. it was simply dissolved away.

Reply:
I'd actually still prefer if this were moved into Coordinate as described, even if the implantation is much simpler now.

Reason being that `Coordinate` is kind of the currency of lots of geo algorithms, so it'd be great if we could start to leverage this functionality in those tests too.

Reply:
Thank you .. yep I had only sorted this out for RelativeEq... While I was fixing this - I noticed that I was systematically over constraining this stuff with 'Float' ... fixed.

this commit.

Removed type constraint on RelativeEq and AbsDiffEq that the type must be Float. Also AbsDiffEq for Point now uses AbsDiffEq in Coords.